src/ruby/puppetserver-lib/puppet/server/ast_compiler.rb

```diff
@@ -51,11 +51,23 @@ def self.compile_ast(code, compile_options, boltlib_path)
                         compile_options['variables']['values']
                       end
 
+          target_variables = if compile_options.key?('target_variables')
+                               compile_options['target_variables']['values'].each_with_object({}) do |param_hash, acc|
+                                 acc[param_hash.keys.first] = param_hash.values.first
+                               end
+                             else
+                               {}
+                             end
+
+          variables = {
```

Comment:
Inspired by https://github.com/puppetlabs/bolt/pull/2277

Reply:
This is fine. We could nitpick on the names but this is what bolt has done.